blocks/editable/patterns.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * External dependencies
+ */
+import tinymce from 'tinymce';
+import { find, get, escapeRegExp, partition, drop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { ESCAPE, ENTER, SPACE, BACKSPACE } from 'utils/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import { getBlockTypes } from '../api/registration';
+
+/**
+ * Browser dependencies
+ */
+const { setTimeout } = window;
+
+export default function( editor ) {
+	const getContent = this.getContent.bind( this );
+	const { onReplace } = this.props;
+
+	const VK = tinymce.util.VK;
+	const settings = editor.settings.wptextpattern || {};
+
+	const patterns = getBlockTypes().reduce( ( acc, blockType ) => {
+		const transformsFrom = get( blockType, 'transforms.from', [] );
+		const transforms = transformsFrom.filter( ( { type } ) => type === 'pattern' );
+		return [ ...acc, ...transforms ];
+	}, [] );
+
+	const [ enterPatterns, spacePatterns ] = partition(
+		patterns,
+		( { regExp } ) => regExp.source.endsWith( '$' ),
```

Comment:
ES2015 string prototypes methods aren't polyfilled, so this will error in IE11.

See also: #746